api-client/aws-fetch.ts

```diff
@@ -0,0 +1,105 @@
+import AsyncStorage from "@react-native-async-storage/async-storage"
+import { CognitoUser, CognitoUserSession } from "amazon-cognito-identity-js"
+import { CognitoIdentityCredentials } from "aws-sdk"
+
+const { AWS_REGION, COGNITO_USER_POOL_ID, IDENTITY_POOL_ID } = process.env
+
+const loginsKey = `cognito-idp.${AWS_REGION}.amazonaws.com/${COGNITO_USER_POOL_ID}`
+
+export interface AuthDetails {
+  isLoggedIn: boolean
+  jwtToken: string | null
+  awsCredentials: CognitoIdentityCredentials | null
+  cognitoUser: CognitoUser | null
+}
+
+// refresh cognito token if expired
+async function refreshToken (callback?: () => void): Promise<void> {
+  try {
+    const storedAuthDetails = await AsyncStorage.getItem("cognitoAuth")
+    if (storedAuthDetails) {
+      globalThis.cognitoAuth = JSON.parse(storedAuthDetails)
+    }
```

Comment:
Things like auth tokens are probably not best in async storage since they are private credentials, but rather the keychain (or whatever android calls it I'm not sure) since it's encrypted.

[This](https://github.com/oblador/react-native-keychain) looks a like a pretty decent library to facilitate that.